### PR TITLE
Adds additional options for ingest requests

### PR DESCRIPTION
**Pull request by cwingard**
Adds stage and mock as ingest request types and adds ability to specify a starting and/or ending date and time to bound the ingests (based on instrument time, not file modification time).

### CODE REVIEW

**Review by crisien — DISMISSED** [review]

File: ingestions/ingest_csvs.py [hunk 1]
```diff
@@ -136,15 +138,20 @@ def main(argv=None):
 
     # assign input arguments.
     parser.add_argument("-c", "--csvfile", dest="csvfile", type=Path, required=True)
-    parser.add_argument("-t", "--ingest_type", dest="ingest_type", type=str, choices=('recovered', 'telemetered'),
+    parser.add_argument("-t", "--ingest_type", dest="ingest_type", type=str, 
+                        choices=('recovered', 'telemetered', 'stage', 'mock'),
```
Comment (cdobs) [inline ingestions/ingest_csvs.py]: Stage and mock should be options for "state" within the ingest request. Options for type are recovered, telemetered, and playback. Options for state are stage, mock, run.

**Review by cdobs — CHANGES_REQUESTED** [review]
Changes to beginData/endData look good but the code should be updated to add mock and stage as options for "state" within the ingest request and not "type"

File: ingestions/ingest_csvs.py (hunk 1 above)
Comment (cwingard) [reply ingestions/ingest_csvs.py]: OK, these have been properly split out, with run set as the default state.

**Review by cdobs — APPROVED** [review]
All changes look good. Tested script with new flags and it works great. Thanks Chris!

**Review by crisien — APPROVED** [review]

**Pull request merged**